test/api/v3/unit/models/group.test.js

```diff
@@ -1418,6 +1418,42 @@ describe('Group Model', () => {
         expect(updatedParticipatingMember.achievements.lostMasterclasser).to.not.eql(true);
       });
 
+      it('gives out super awesome Masterclasser achievement even when quests done out of order', async () => {
+        quest = questScrolls.lostMasterclasser1;
+        party.quest.key = quest.key;
+
+        questLeader.achievements.quests = {
+          mayhemMistiflying1: 1,
+          mayhemMistiflying2: 1,
+          mayhemMistiflying3: 1,
+          stoikalmCalamity1: 1,
+          stoikalmCalamity2: 1,
+          stoikalmCalamity3: 1,
+          taskwoodsTerror1: 1,
+          taskwoodsTerror2: 1,
+          taskwoodsTerror3: 1,
+          dilatoryDistress1: 1,
+          dilatoryDistress2: 1,
+          dilatoryDistress3: 1,
+          lostMasterclasser2: 1,
+          lostMasterclasser3: 1,
+          lostMasterclasser4: 1,
+        };
+        await questLeader.save();
+        await party.finishQuest(quest);
+
+        let [
+          updatedLeader,
+          updatedParticipatingMember,
+        ] = await Promise.all([
+          User.findById(questLeader._id),
```

Comment:
findById() should be followed by `.exec()` otherwise it returns an object that has a `.then` but is not a realy promise, see http://mongoosejs.com/docs/api.html#model_Model.findById

Reply:
Okay, thanks for the feedback. Got that changed.